### DISCUSSION TITLE
Tasks - do they support arg from user input ?

### DISCUSSION THREAD

**Discussion by coolemur** [zed-industries/zed#10823] [Feature Requests]
For example something like:

```
  {
    "label": "test",
    "args": ["arg1"],
    "command": "echo \"Hello ${arg1} World\""
  },
```

When running task 'test', execution would wait for user input, after entering it it would output "Hello <something that user typed> world" 

Asking this, because I am not sure how args are currenlty working (didn't find much documentation about it). But from first look it looks like arg is just something that gets appended at the end of command as argument

But what is the point of this if you have to edit task json to change the arg ?

Am I missing something here ?

**Comment by coolemur** [comment]
Example from VSCode of what I am trying to achieve using Tasks:

```
"kill port": "kill -9 $(lsof -i tcp:${input:PORT} -t) 2> /dev/null"
```

I want PORT to be inserted by user on task run

**Comment by SomeoneToIgnore** [comment]
There are no support for user arguments (yet?), so Zed is not capable to wait for any extra input from the user.

Yet tasks themselves are templates, and Zed provides a set of template variables to use, automatically substituting those before running: https://zed.dev/docs/tasks

That might help a bit in your case, maybe.

**Reply by coolemur** [reply]
Variables doesn't help much when there is no port listed anywhere in zed editor in this case
I figured to use as many Tasks as there are ports (doesn't look nice, but that's something)
Might be something for a new feature for zed (user input in Tasks as variable)

**Reply by SomeoneToIgnore** [reply]
> use as many Tasks as there are ports

You could use `read` in a custom script and invoke that via a task, that looks much better than multiple files at least.
I would imagine, it could be possible to write a spell with `read ; command` directly inside the tasks.json in Zed, but have not tried it.

> a new feature for zed (user input in Tasks as variable)

Yes, feel free to describe it in an issue if feels bad even after the `read` custom script — I do not think we will work on custom user input features in the near future, but somebody from the community might notice and improve it.
